avro-converter/src/test/java/io/confluent/connect/avro/AvroConverterTest.java

```diff
@@ -135,4 +154,68 @@ public void testNull() {
     SchemaAndValue schemaAndValue = converter.toConnectData(TOPIC, converted);
     assertEquals(SchemaAndValue.NULL, schemaAndValue);
   }
+
+  @Test
+  public void testVersionExtracted() throws Exception {
+    // Version info should be extracted even if the data was not created with Copycat. Manually
+    // register a few compatible schemas and validate that data serialized with our normal
+    // serializer can be read and gets version info inserted
+    KafkaAvroSerializer serializer = new KafkaAvroSerializer(schemaRegistry);
+
+    // Pre-register to ensure ordering
+    org.apache.avro.Schema avroSchema1 = org.apache.avro.SchemaBuilder
+        .record("Foo").fields()
+        .requiredInt("key")
+        .endRecord();
+    schemaRegistry.register(TOPIC + "-value", avroSchema1);
+
+    org.apache.avro.Schema avroSchema2 = org.apache.avro.SchemaBuilder
+        .record("Foo").fields()
+        .requiredInt("key")
+        .requiredString("value")
+        .endRecord();
+    schemaRegistry.register(TOPIC + "-value", avroSchema2);
+
+
+    // Get serialized data
+    org.apache.avro.generic.GenericRecord avroRecord1
+        = new org.apache.avro.generic.GenericRecordBuilder(avroSchema1).set("key", 15).build();
+    byte[] serializedRecord1 = serializer.serialize(TOPIC, avroRecord1);
+    org.apache.avro.generic.GenericRecord avroRecord2
+        = new org.apache.avro.generic.GenericRecordBuilder(avroSchema2).set("key", 15).set
+        ("value", "bar").build();
+    byte[] serializedRecord2 = serializer.serialize(TOPIC, avroRecord2);
+
+
+    SchemaAndValue converted1 = converter.toConnectData(TOPIC, serializedRecord1);
+    assertEquals(1L, (long) converted1.schema().version());
+
+    SchemaAndValue converted2 = converter.toConnectData(TOPIC, serializedRecord2);
+    assertEquals(2L, (long) converted2.schema().version());
+  }
+
+
+  @Test
+  public void testVersionMaintained() {
+    // Version info provided from the Copycat schema should be maintained. This should be true
```

Comment:
Good testing strategy. This makes sure that the schema version is kept no matter what the state of the schema registry is.